src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+  const sum = (num) => {
+    if (num === undefined) {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    }
+    total += num;
+
+    return sum;
```

Comment:
Consider adding input validation to ensure that `num` is a number. This will prevent unexpected behavior if a non-numeric value is passed to the `sum` function.